libs/MeshKernel/include/MeshKernel/AveragingStrategies/AveragingStrategy.hpp

```diff
@@ -36,6 +36,9 @@ namespace meshkernel::averaging
     public:
         virtual ~AveragingStrategy() = default;
 
+        /// @brief Reset the state of the averaging strategy, ready for the next calculation.
+        virtual void Reset(const Point& interpolationPoint) = 0;
+
         /// @brief Adds the specified sample point.
         /// @param[in] samplePoint The sample point to add to this strategy.
         /// @param[in] sampleValue The value associated with the sample point.
```

Comment:
Never liked the how samplePoint is a dummy arg in most derived classes.  Only `ClosestAveragingStrategy` and `InverseWeightedAveragingStrategy` use it. Having a SetSamplePoint method in the base is also a bad idea because again, only used when `ClosestAveragingStrategy` and `InverseWeightedAveragingStrategy` arte created.  
Can you think of a better way to do this?